Automattic/wp-calypso
# Stats: Apply dropdown navigation for small screen on summary pages

#### Proposed Changes

* Creare new component for swapping different components at a specified breakpoint
* Swap tabs for a dropdown for small screens on a summary page

#### Testing Instructions



* Go to stats and navigate to `View all`/`View details` in a footer of a module below the chart (you will go to a summary detail page)
* apply the feature flag by adding `?flags=stats/horizontal-bars-everywhere` to the URL
* Resize the page and make sure that for a small screen, tabs are converted to a dropdown
* Bonus test: Make sure that the navigation doesn't break for different screen sizes

Widescreen header:
![Screenshot 2023-01-25 at 7 29 34 PM](https://user-images.githubusercontent.com/112354940/214496330-3e4ba7b8-5c8b-4cd7-8bb7-3f24f80a46a1.png)

Mobile screen header:
![Screenshot 2023-01-25 at 7 29 56 PM](https://user-images.githubusercontent.com/112354940/214496382-7f5d58cc-f3f5-473e-9c4e-e739812ad33e.png)

Before the proposed changes, the widescreen header was applied to all breakpoints. This PR adds the dropdown visible on the mobile screenshot. 

#### Pre-merge Checklist





Related to #71018
